### PR TITLE
Migrate query source to React: Set of updates

### PR DESCRIPTION
**Pull request by kravets-levko**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

- [x] Refactor: get rid of `Title` service + directive; replace with `document.title`;
- [x] Manage permissions dialog;
- [x] Show API key dialog;
- [x] Refactor: extract actions to publish/unpublish/rename query.

## Related Tickets & Documents

getredash/redash#4429

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

image

### CODE REVIEW

File: client/app/pages/dashboards/dashboard.js [hunk 1]
```diff
@@ -172,7 +171,8 @@ function DashboardCtrl(
   };
 
   const renderDashboard = (dashboard, force) => {
-    Title.set(dashboard.name);
+    // TODO: Title should be updated after editing as well
```
Comment (gabrieldutra) [inline client/app/pages/dashboards/dashboard.js]: 2a2bfb81c66bdfb30cc1727a72dc7ec292a9aeca

File: client/app/components/queries/ApiKeyDialog/index.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                Regenerate
+              </Button>
+            )}
+          </Input.Group>
+        </div>
+
+        <h5>Example API Calls:</h5>
+        <div>
+          Results in CSV format:
+          <pre>{csvUrl}</pre>
```
Comment (gabrieldutra) [inline client/app/components/queries/ApiKeyDialog/index.jsx]: Good use case for `InputWithCopy` or `CodeBlock`? (I think `CodeBlock` -- it's used in the Embed Query dialog too)

Comment (kravets-levko) [reply client/app/components/queries/ApiKeyDialog/index.jsx]: Brilliant idea!

File: client/app/pages/dashboards/dashboard.js (hunk 1 above)
Comment (kravets-levko) [reply client/app/pages/dashboards/dashboard.js]: Yes, I saw it few minutes ago; this comment I added this morning :grin: Let it stay here, anyway we'll remove this file soon

File: client/app/components/queries/ApiKeyDialog/index.jsx (hunk 2 above)
Comment (kravets-levko) [reply client/app/components/queries/ApiKeyDialog/index.jsx]: Done:

image

**Pull request merged**